WcaOnRails/lib/gsuite_mailing_lists.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+ENV['GOOGLE_APPLICATION_CREDENTIALS'] = Rails.root.join("../secrets/application_default_credentials.json").to_s
+
+require 'googleauth'
+require 'googleauth/stores/file_token_store'
+require 'google/apis/admin_directory_v1'
+
+module GsuiteMailingLists
+  def self.sync_group(group, desired_emails)
+    service = get_service
+
+    desired_emails = desired_emails.map do |email|
+      if email.include?("+")
+        old_email = email
+        email = email.gsub(/\+[^@]*/, '')
+        puts "Warning: '#{old_email}' contains a plus sign, and google groups seems to not support + signs in email addresses, so we're going to add '#{email}' instead."
+      end
+      email
+    end
+
+    members = service.fetch_all(items: :members) do |token|
+      service.list_members(group, page_token: token)
+    end
+    current_emails = members.map(&:email)
+
+    emails_to_remove = current_emails - desired_emails
+    emails_to_add = desired_emails - current_emails
+
+    # First, remove all the emails we don't want.
+    emails_to_remove.each do |email|
+      service.delete_member(group, email)
+    end
+
+    # Last, add all the emails we do want.
+    emails_to_add.each do |email|
+      new_member = Google::Apis::AdminDirectoryV1::Member.new(email: email)
```

Comment:
Can we add here the name too? If not, it will be very confusing knowing who is who when they send emails. It would be nice also assigning to Senior Delegates the role of Manager.

Reply:
I don't see any way to set the "display name" for a group member via the UI or via the API (https://developers.google.com/admin-sdk/directory/v1/guides/manage-group-members#create_member).